### DISCUSSION TITLE
iframes[0].content_document in with selenium_driverless.sync?

### DISCUSSION THREAD

**Discussion by Fragaile** [kaliiiiiiiiii/Selenium-Driverless#78] [Q&A] [answered]
Is there any way to interact with an iframe with selenium_driverless.sync? There seems to be no function ``.content_document`` for web elements with the sync version or am I missing something? Thanks!

Code:
```
from selenium_driverless.sync import webdriver
from selenium_driverless.types.by import By

options = webdriver.ChromeOptions()
driver = webdriver.Chrome(options = options)

driver.get('https://www.google.com/')
iframes = driver.find_elements(By.TAG_NAME, "iframe")
document = iframes[0].content_document
```

Error: ``RuntimeWarning: coroutine 'WebElement.content_document' was never awaited``

**Comment by ttlns** [comment]
Uh that's interesting. I'll give it a try. Technically, the syncification should handle that automatically👀

**Comment by Fragaile** [comment]
@kaliiiiiiiiii Yeah that's what I assumed at first too.. Also is there any way to wait for an element to appear? Or can I just do timeout = ...?
Thanks!

**Reply by ttlns** [reply]
1. https://github.com/kaliiiiiiiiii/Selenium-Driverless/commit/9f2723caf338b148fa2b3ba8b7bb101f11e1c182 should fix this, thanks for reporting that bug! Basically, I missed passing that a `WebElement` should be sync at deserialisation. And yes, you can actually do:
 ```python
 body_elem = driver.execute_script("return document.body // or whatever element")
 ```
2. `driver.find_element(By.XPATH, "some_xpath", timeout=10)` for example should work just fine.

**Comment by ttlns** [comment]
Tested with:
```python
from selenium_driverless.sync import webdriver
from selenium_driverless.types.by import By

options = webdriver.ChromeOptions()
with webdriver.Chrome(options=options) as driver:
    driver.get('https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_iframe', wait_load=False)
    iframe = driver.find_element(By.TAG_NAME, "iframe", timeout=10)
    document = iframe.content_document
    driver.quit(clean_dirs=True)
```

Now and seems to be fixed => yanking current version and releasing a patch now.

**Reply by Fragaile** [reply]
That was so quick :D, appreciate it! ❤️

**Reply by ttlns** [reply, marked as answer]
Yep, don't like to release versions which break previous stuff:/
 
 https://pypi.org/project/selenium-driverless/1.6.1/ Released now btw

**Reply by ttlns** [reply]
Feel free to confirm if it's working now and then close this discussion:)

**Reply by Fragaile** [reply]
Problem to new release within 40 minutes.. that's impressive.

**Reply by Fragaile** [reply]
Works great!
